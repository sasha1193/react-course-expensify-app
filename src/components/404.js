import React from 'react';
import {Link} from 'react-router-dom';


const NotFound = () => (
    <div>
        404 - <Link to="/">Go HOME</Link>
    </div>
);


export default NotFound;