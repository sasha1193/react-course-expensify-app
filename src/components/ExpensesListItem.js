import React from 'react';
import moment from 'moment';
import { NavLink } from 'react-router-dom';


const ExpensesListItem = ({ id, description, amount, createAt }) => (
    <div>
        <NavLink to={`/edit/${id}`}><h3>{description}</h3></NavLink>
        <p>{amount} - {moment(createAt).format('MMM Do YYYY')}</p>
    </div>
);



export default ExpensesListItem;