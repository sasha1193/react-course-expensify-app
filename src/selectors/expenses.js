//Get visible expenses
import moment from "moment";


export default (expenses, { text, sortBy, startDate, endDate }) => {
    return expenses.filter((expense) => {
        const momentCreatedAt = moment(expense.createAt)
        const startDateMatch = startDate ? startDate.isSameOrBefore(momentCreatedAt, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(momentCreatedAt, 'day') : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if(sortBy === 'date') {
            return a.createAt < b.createAt ? 1 : -1;
    } else if (sortBy === 'amount') {
            return a.amount < b.amount ? 1 : -1;
    }
    })
};

