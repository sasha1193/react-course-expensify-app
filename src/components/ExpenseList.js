import React from 'react';
import ExpensesListItem from './ExpensesListItem';
import getVisibleExpenses from '../selectors/expenses';
import { connect } from 'react-redux';

export const ExpensesList = (props) => (
    <div>
    {
        props.expenses.length === 0 ? (
            <p>No expenses</p>
        ) : (
            props.expenses.map((expense) => 
            <ExpensesListItem {...expense} key={expense.id} />    
            )
        )
    }
        
    </div>
);

const mapStateToProps = (state)=> {
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    };
}


export default connect(mapStateToProps)(ExpensesList);


