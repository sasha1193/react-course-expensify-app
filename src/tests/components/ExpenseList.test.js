import React from 'react';
import {shallow} from 'enzyme';
import { ExpensesList } from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';


test('should render ExpenseList with expenses', () => {
    const wrapper = shallow(<ExpensesList expenses={expenses} />)
    expect(wrapper).toMatchSnapshot();
})

test('should render ExpenseList with empty message', () => {
    const wrapper = shallow(<ExpensesList expenses={[]} />) 
    expect(wrapper).toMatchSnapshot();
})
