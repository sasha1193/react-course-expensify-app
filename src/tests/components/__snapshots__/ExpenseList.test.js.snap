// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <ExpensesListItem
    amount={195}
    createAt={0}
    description="Gum"
    id="1"
    note="some note"
  />
  <ExpensesListItem
    amount={1200}
    createAt={-345600000}
    description="Rent"
    id="2"
    note="other note"
  />
  <ExpensesListItem
    amount={12000}
    createAt={345600000}
    description="Credit Card"
    id="3"
    note="third note"
  />
</div>
`;
